src/components/atom_preference_language_checkbox/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { putPreferenceApi } from '@/api/preferences/put-preference.api'
+import { getLanguageFullName } from '@/global.constants'
+import { GlobalLanguageCode } from '@/global.interface'
+import { preferenceState } from '@/recoil/preferences/preference.state'
+import { Checkbox, FormControlLabel } from '@mui/material'
+import { FC } from 'react'
+import { useRecoilCallback, useRecoilValue, useSetRecoilState } from 'recoil'
+
+interface Props {
+  languageCode: GlobalLanguageCode
+}
+
+const PreferenceLanguageCheckbox: FC<Props> = ({ languageCode }) => {
+  const putPreferenceState = useSetRecoilState(preferenceState)
+  const preference = useRecoilValue(preferenceState)
+  const onChange = useRecoilCallback(
+    ({ set, snapshot }) =>
+      async (_, checked: boolean) => {
+        const preference = await snapshot.getPromise(preferenceState)
+        if (!preference) return
+
+        const nativeLanguagesSet = new Set([
+          ...preference.nativeLanguages,
+          languageCode,
+        ])
+
+        if (!checked) nativeLanguagesSet.delete(languageCode)
```

Comment:
```suggestion
        const nativeLanguagesSet = new Set(preference.nativeLanguages)
        if (checked) nativeLanguagesSet.add(languageCode)
        else nativeLanguagesSet.delete(languageCode)
```

Reply:
handled in https://github.com/ajktown/wordnote/pull/101/commits/cb16766dcdf640d210b547dbf382d85e6d582726